src/nick.c

```diff
@@ -34,7 +34,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 User *find_account(const Nick *nptr)
 {
-    return *nptr->svid ? find_user(nptr->svid) : find_user(nptr->nick);
+    return nptr->svid != "0" ? find_user(nptr->svid) : find_user(nptr->nick);
```

Comment:
This will not work as it will compare addresses. You need to do strcmp() to compare nptr->svid against "0".

Reply:
Resolving, newer comment below.